ds3-autogen-parser/src/main/java/com/spectralogic/ds3autogen/converters/ResponseTypeConverter.java

```diff
@@ -201,13 +201,36 @@ protected static Ds3Type toDs3Type(final EncapsulatingTypeNames encapsulatingTyp
                                 new Ds3AnnotationElement("Value", getAnnotationName(encapsulatingType), "java.lang.String"))));
         return new Ds3Type(
                 encapsulatingType.getSdkName() + NAMESPACE_ARRAY_RESPONSE_TYPES,
+                getNameToMarshalForEncapsulatingType(encapsulatingType),
                 ImmutableList.of(
                         new Ds3Element(
                                 English.plural(stripPath(encapsulatingType.getSdkName())),
                                 "array",
                                 encapsulatingType.getSdkName(),
                                 annotations,
-                                false)));
+                                false)),
+                null);
+    }
+
+    /**
+     * Determines the name to marshal value for the encapsulating type.
+     * Note: NameToMarshal = "" denotes no encapsulating tag
+     * NameToMarshal = null denotes 'Data' encapsulating tag
+     */
+    protected static String getNameToMarshalForEncapsulatingType(final EncapsulatingTypeNames encapsulatingType) {
```

Comment:
This seems like we should use a type to encapsulate the encapsulating type rather than relying on nulls and empty strings.